### DISCUSSION TITLE
C# syntax highlighting is completely broken?

### DISCUSSION THREAD

**Discussion by EgorBo** [github-linguist/linguist#6412] [Classification] [answered]
I'm sorry if this repo is completely unrelated, but I don't know where to start. It seems that C# syntax highlighting has been completely broken for at least a month at github and I'm struggling to understand why.

Consider a very simple function:
```csharp
int Add(int a, int b) => a + b;
```
Yes, it's wrapped with '''csharp, now let me do the same with '''cpp
```cpp
int Add(int a, int b) => a + b;
```

so C# is so broken that I now use C++ for my C# snippets 🙁. This might be something related to top-level statements, let me try a class now:
```csharp
class P { int Add(int a, int b) => a + b; }
```
still grey (`class=storage-modifier-import`) for keywords... why?

**Comment by lildude** [comment, marked as answer]
This is indeed the wrong place as it is a problem with the upstream tree-sitter grammar but I can confirm an update is coming in the next week or two which should fix the issue. If not, [GitHub support](https://support.github.com/?tags=dotcom-footer) is the best place to bring this up as Linguist is not responsible for the C# grammar.
